libs/dtm/dtm_recovery.py

```diff
@@ -134,28 +141,115 @@ def perform_ops(self, workload_info: list, queue, skipread: bool = True,
+
+            self.log.info("Process %s restarted successfully", process)
+
+        self.log.info("Polling hctl status to check if all services are online")
+        resp = ha_obj.poll_cluster_status(pod_obj=master_node, timeout=300)
+        return resp
+
+    def get_process_state(self, master_node, pod_name, container_name, process_name,
+                          process_ids: list = None):
+        """
+        Function to get given process state
+        :param master_node: Object of master node
+        :param pod_name: Name of the pod on which container is residing
+        :param container_name: Name of the container inside which process is running
+        :param process_name: Name of the process
+        :param process_ids: List of Process IDs
+        :return: bool, dict
+        e.g. (True, {'0x19': 'M0_CONF_HA_PROCESS_STARTED', '0x28': 'M0_CONF_HA_PROCESS_STARTED'})
+        """
+        process_state = dict()
+        self.log.info("Get processes running inside container %s of pod %s", container_name,
+                      pod_name)
+        resp = master_node.get_all_container_processes(pod_name=pod_name,
+                                                       container_name=container_name)
+        if process_ids:
+            self.log.info("Extract list of %s processes having IDs %s", process_name, process_ids)
+            process_list = [(ele, p_id) for ele in resp for p_id in process_ids if p_id in ele
+                            and process_name in ele]
+            if len(process_ids) != len(process_list):
+                return False, f"All process IDs {process_ids} are not found. " \
+                              f"All processes running in container are: {resp}"
+        else:
+            self.log.info("Extract list of %s processes", process_name)
+            process_list = [ele for ele in resp if process_name in ele]
+        compile_exp = re.compile('"state": "(.*?)"')
+        for i_i in process_list:
+            process_state[i_i[1]] = compile_exp.findall(i_i[0])[0]
+
+        return True, process_state
+
+    def poll_process_state(self, master_node, pod_name, container_name, process_name,
+                           process_ids: list = None, status: str = expected_process_state,
+                           timeout: int = 300):
+        """
+        Helper function to poll the process states
+        :param master_node: Object of master node
+        :param pod_name: Name of the pod on which container is residing
+        :param container_name: Name of the container inside which process is running
+        :param process_name: Name of the process
+        :param process_ids: List of Process IDs
+        :param status: Expected status of process
+        :param timeout: Poll timeout
+        :return: Bool
+        """
+        resp = False
+        self.log.info("Polling process states")
+        start_time = int(time.time())
+        while timeout > int(time.time()) - start_time:
+            time.sleep(60)
```

Comment:
what is expected time for process state chnage? Do we need to wait for 60 secs? @priyankaborawake  @ApurwaMohite

Reply:
@sonalk0209 This polling is done for dtm recovery. 
 As we are testing process restart, the dtm recovery is not expected to take much time. Hence kept timeout to 5 mins.